jena-kafka-connector/src/main/java/org/apache/jena/kafka/common/FusekiProjector.java

```diff
@@ -263,7 +316,7 @@ protected void commitTransactionIfNeeded(Event<Bytes, RdfPayload> event) {
             // No batching enabled, always commit immediately
             FusekiKafka.LOG.trace("[{}] Committing due to batching disabled", this.topicNames);
             this.commit();
-        } else if (this.eventsSinceLastCommit.size() >= this.batchSize) {
+        } else if (!this.highLagDetected && this.eventsSinceLastCommit.size() >= this.batchSize) {
```

Comment:
- [ ] Update comments below as to why we only do this when High Lag isn't detected